deepdow/utils.py

```diff
@@ -256,8 +262,10 @@ def raw_to_Xy(raw_data, lookback=10, horizon=10, gap=0, freq='B', included_asset
 
     index = pd.date_range(start=raw_data.index[0], end=raw_data.index[-1], freq=freq)
 
+    # ? Maybe pass columns and raw_data.values to avoid sticky Frozen Lists
```

Comment:
Can you maybe elaborate on this one?

Reply:
MultiIndex seems to rely on Frozen Lists, which gave me some problems when I didn't pass the data.values instead of the data as a frame to create a new dataframe. The solution is so simple it's not worth a discussion, I'll do the fix.

Reply:
Actually, if all this is doing is upsampling the dataset with a specified `freq` and filling nans with `.ffill().bfill()`, maybe `raw_data.resample` could do the same with less code...

Reply:
Yeh, I guess `resample` could be also used:) The principal goal is to make sure that there are no NaNs in the data and that the frequency is fixed.